numba_dpex/core/datamodel/models.py

```diff
@@ -55,58 +55,68 @@ def __init__(self, dmm, fe_type):
         super(GenericPointerModel, self).__init__(dmm, fe_type, be_type)
 
 
-class USMArrayModel(StructModel):
-    """A data model to represent a Dpex's array types in LLVM IR.
+class USMArrayDeviceModel(StructModel):
+    """A data model to represent a usm array types in LLVM IR on device.
 
     Dpex's ArrayModel is based on Numba's ArrayModel for NumPy arrays. The
-    dpex model adds an extra address space attribute to all pointer members
-    in the array.
+    dpex model adds an extra address space attribute to almost all pointer
```

Comment:
The first two paragraphs in the docstring may be updated as follows. Basically, let us be explicit with the details.

>The ``USMArrayDeviceModel`` adds an extra address space attribute to the ``data`` member. The extra attribute is needed when passing usm_ndarray array arguments to kernels that are compiled for certain OpenCL GPU devices. Note that the address space attribute is applied only to the ``data`` member and not other members of ``USMArrayDeviceModel`` that are pointers. It is done this way as other pointer members such as ``meminfo`` are not used inside a kernel and these members maybe removed from the ``USMArrayDeviceModel`` in future (refer #929).
>
> We use separate data models for host (``USMArrayHostModel``) and device (``USMArrayDeviceModel``) as the address space attribute is only required for kernel functions and not needed for functions that are compiled for a host memory space.